src/index.ts

```diff
@@ -11,15 +22,31 @@ import ErrorScript from './scripts/error';
 import SettingsScript from './scripts/settings';
 
 import { AppVeyor } from './lib/appveyor';
+import { SecureBrain } from './lib/secure_brain';
 import { Config } from './lib/config';
 
 module.exports = (robot: Robot) => {
+
+  const requiredVariables = [
+    'APPVEYOR_ACCOUNT',
+    'APPVEYOR_WEBHOOK_TOKEN',
+    'SECURE_BRAIN_KEY'
+  ];
+
+  requiredVariables.forEach(e => {
+    if (process.env[e] === undefined) {
+      console.error(`Please define the environment variable for ${e}`);
+      process.exit(1);
+    }
+  });
+
   const appveyor = new AppVeyor(robot.http.bind(robot), Config.appveyor.account);
+  const secureBrain = new SecureBrain(robot, Config.secure_brain.key);
 
   ErrorScript(robot);
   HelloScript(robot);
-  BuildScript(robot, appveyor);
-  BuildsScript(robot, appveyor);
-  DeployScript(robot, appveyor);
-  SettingsScript(robot);
+  BuildScript(robot, appveyor, secureBrain);
+  BuildsScript(robot, appveyor, secureBrain);
+  DeployScript(robot, appveyor, secureBrain);
+  SettingsScript(robot, secureBrain);

```

Comment:
Yea the first four read the token, the last one sets it. You think it should be wrapped in a token provider or something?